libs/core/src/lib/engine/interfaces.ts

```diff
@@ -52,11 +52,11 @@ export namespace FlogoAppModel {
     settings: Settings;
     action: {
       ref: string;
-      settings: Settings;
-      input: {
+      settings?: Settings;
```

Comment:
settings, input and output are optional in the [app schema](https://github.com/project-flogo/core/blob/master/internal/schema/schema.json)